pkg/features/builder.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package features
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/e2e-framework/pkg/internal/types"
+)
+
+// FeatureBuilder represents is a type to define a
+// testable feature
+type FeatureBuilder struct {
+	feat *defaultFeature
+}
+
+func New(name string) *FeatureBuilder {
+	return &FeatureBuilder{feat: newDefaultFeature(name)}
+}
+
+// WithLabel adds a test label key/value pair
+func (b *FeatureBuilder) WithLabel(key, value string) *FeatureBuilder {
+	b.feat.labels[key] = value
+	return b
+}
+
+// Setup adds a new setup step that will be applied prior to feature test.
+func (b *FeatureBuilder) Setup(fn Func) *FeatureBuilder {
+	b.feat.steps = append(b.feat.steps, newStep(fmt.Sprintf("%s-setup", b.feat.name), types.LevelSetup, fn))
+	return b
+}
+
+// Teardown adds a new teardown step that will be applied after feature test.
+func (b *FeatureBuilder) Teardown(fn Func) *FeatureBuilder {
+	b.feat.steps = append(b.feat.steps, newStep(fmt.Sprintf("%s-teardown", b.feat.name), types.LevelTeardown, fn))
+	return b
+}
+
+// Assess adds an assessment step to the feature test.
+func (b *FeatureBuilder) Assess(desc string, fn Func) *FeatureBuilder {
```

Comment:
nit: I'm used to the term `assert` from other programming languages. Not sure if it's been considered, but perhaps a discussion opportunity?

Reply:
@wilsonehusin I'm waiting this first PR to get in then we can setup golangci-lint and tests and other things

Reply:
@wilsonehusin yeah I thought about assert. But wanted a term that is more than assertion, but an actual assessment.  Maybe Test could be a better term.  Once it's merged we can change.